city_scrapers/spiders/il_liquor_control.py

```diff
@@ -0,0 +1,161 @@
+import datetime
+import requests
+
+from city_scrapers_core.constants import BOARD
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class IlLiquorControlSpider(CityScrapersSpider):
+    name = "il_liquor_control"
+    agency = "Illinois Liquor Control Commission"
+    timezone = "America/Chicago"
+    allowed_domains = ["www2.illinois.gov"]
+    start_urls = [
+        "https://www2.illinois.gov/ilcc/Divisions/Pages/Legal/"
+        "Hearing-Schedule-for-Chicago-IL-and-Springfield-IL.aspx",
+        "https://www2.illinois.gov/ilcc/Divisions/Pages/Legal/Meeting-minutes.aspx"
+    ]  # start_urls[0] contains links future meetings. start_urls[1] contains past
+    # meeting minutes links
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+
+        Each tentative meeting has own page. start_urls[0] is only used to get links for the pages.
+        The links are sent to _next_meeting.
+
+        start_urls[1] is for obtaining datetime info for past meetings. Then checks if
+        meeting detail page is still valid. If valid scrape through _prev_meeting_w_page. If not
+        valid, scrape available info from _prev_meeting_wo_page.
+        """
+
+        if response.url == self.start_urls[0]:
+            for future_meeting_href in response.css('div.soi-link-item a::attr(href)').extract():
+                yield response.follow(future_meeting_href, callback=self._next_meeting)
+
+        if response.url == self.start_urls[1]:
+            for idx, past_meeting_date in enumerate(response.css("div.link-item ::text").extract()):
```

Comment:
A simpler way of doing this might be to iterate over the links and pull out the link text inside the loop, rather than re-parse the response each time. Something like

```python
for past_meeting in response.css("div.link-item a"):
    past_meeting_text = past_meeting.css("*::text").extract_first()
    minutes_href = past_meeting.attrib["href"]
    ...
```

Also, it looks like they only started publishing the detail pages in 2019, so we can parse the year out of the date here and exit out of the loop if the meeting date is before then. We've handled this differently on different spiders, but generally it's easier to only parse meetings within the past year on a consistent basis because we can pull all old documents in a single archival run